metals/src/main/scala/scala/meta/internal/metals/callHierarchy/OutgoingCallsFinder.scala

```diff
@@ -0,0 +1,252 @@
+package scala.meta.internal.metals.callHierarchy
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Member
+import scala.meta.Name
+import scala.meta.Pat
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.DefinitionProvider
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ReferenceProvider
+import scala.meta.internal.metals.ResolvedSymbolOccurrence
+import scala.meta.internal.mtags.Semanticdbs
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.semanticdb.IdTree
+import scala.meta.internal.semanticdb.OriginalTree
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.SelectTree
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j
+import org.eclipse.lsp4j.Position
+
+class OutgoingCallsFinder(
+    semanticdbs: Semanticdbs,
+    definition: DefinitionProvider,
+    references: ReferenceProvider,
+    trees: Trees,
+)(implicit ec: ExecutionContext)
+    extends CallHierarchyHelpers {
+
+  private case class DefinitionInformation(
+      occurence: SymbolOccurrence,
+      source: AbsolutePath,
+      doc: TextDocument,
+  ) {
+    def symbolLength: Int = occurence.symbol.length()
+
+    def definitionTree: Option[Tree] = for {
+      range <- occurence.range
+      from <- trees.findLastEnclosingAt(source, range.toLsp.getStart())
+      definitionTree <- findDefinition(
+        from
+      )
+    } yield definitionTree.root
+
+    def toOutgoingCallResult(
+        fromRange: lsp4j.Range
+    ): Option[FindOutgoingCallsResult] =
+      definitionTree.map(tree =>
+        FindOutgoingCallsResult(
+          occurence,
+          tree.pos.toLsp,
+          List(fromRange),
+          source,
+          doc,
+        )
+      )
+
+    def isSameDefinition(
```

Comment:
Why is this needed? Could you add a sample in a comment? Is it to match the tree with a symbol occurrence?